libcxx/test/libcxx/algorithms/sort_stability.pass.cpp

```diff
@@ -68,8 +68,7 @@ void test_same() {
   auto snapshot_custom_v = v;
   std::sort(v.begin(), v.end());
   std::sort(snapshot_v.begin(), snapshot_v.end());
-  std::sort(snapshot_custom_v.begin(), snapshot_custom_v.end(),
-            [](const EqualType&, const EqualType&) { return false; });
```

Comment:
Why remove this test instead of just adding the new one?

Reply:
The old test case passes `[](const EqualType&, const EqualType&) { return false; }` as a comparator, which is not a "simple comparator", and `__use_branchless_sort` is written in a way that is true only when the `__is_simple_comparator` is true, therefor libc++ chooses to use branchless sort for `std::sort(snapshot_v.begin(), snapshot_v.end());` but not for `std::sort(snapshot_custom_v.begin(), snapshot_custom_v.end(), [](const EqualType&, const EqualType&) { return false; });` but after the change `std::sort(snapshot_custom_v.begin(), snapshot_custom_v.end(), std::less<EqualType>())` uses branchless sort. After this change, both statements will use branchless sort while the main behavior is preserved.

Reply:
But that means we start to strongly write our tests based on our current implementation. That's not a good idea in general. Having some duplicated test coverage can help during refactoring where different paths are taken. So I really like adding a test, I'm just less thrilled with the removal.

Reply:
It's worth noting that this is in `test/libcxx`, not `test/std`, and so there's more lenience for tests that are coupled with the implementation. I do think that we should avoid this practice though.

It's also worth noting that this lambda doesn't meet the requirements of `Compare` because it doesn't impose a [strict weak order](https://eel.is/c++draft/algorithms#alg.sorting.general-4) on its elements, so replacing it with something that does is a good idea.

Reply:
`return false;` imposes a perfectly correct strict weak ordering comparator

Reply:
Also, I still don't understand why we need to change the comparator here. The test checks that randomization happens for any sort you call